### PR TITLE
Housecleaning

### PR DESCRIPTION
**Pull request by ZephrFish**
# Updated:
- Streamlined code to tidy up format functions and convert to python3 format strings, compressed code and made it more efficient.
- Standardised where possible all messages use " instead of '  to make life easier.
- Also flipped the config file to an example and added it to .gitignore, so we don't commit secrets in testing to git :)

### CODE REVIEW

File: credmaster.py [hunk 1]
```diff
@@ -589,33 +582,18 @@ def log_entry(entry):
...
-	ts = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S.%f')[:-3]
-	print('[{}] {}'.format(ts, entry))
+	ts = utc_now.strftime('%Y-%m-%d %H:%M:%S.%f')[:-3]
+	print(f"[{ts}] {entry}")
 
 	if outfile is not None:
-		with open(outfile + "-credmaster.txt", 'a+') as file:
-			file.write('[{}] {}'.format(ts, entry))
-			file.write('\n')
+		with open(outfile, 'a+') as file:
```
Comment (knavesec) [inline credmaster.py:589]: Add back in the "-credmaster.txt" addition to the outfile name, used to distinguish between standard outfile and valid user outfile

File: credmaster.py [hunk 2]
```diff
@@ -541,7 +534,7 @@ def ww_calc_next_spray_delay(offset):
 
 	spray_times = [7,11,15] # launch sprays at 7AM, 11AM and 3PM
 
-	now = datetime.datetime.utcnow() + datetime.timedelta(hours=offset)
+	now = utc_now + datetime.timedelta(hours=offset)
```
Comment (knavesec) [inline credmaster.py:537]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 3]
```diff
@@ -14,12 +13,11 @@
 ]
 
 lock = threading.Lock()
-lock_userenum = threading.Lock()
 q_spray = queue.Queue()
 
 outfile = None
 color = None
-
+utc_now = datetime.datetime.utcnow()
```
Comment (knavesec) [inline credmaster.py:20]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 4]
```diff
@@ -75,26 +74,26 @@ def main(args,pargs):
 
 	# input exception handling
 	if outfile != None:
-		outfile = outfile
-		if os.path.exists(outfile + "-credmaster.txt"):
-			log_entry("File {} already exists, try again with a unique file name".format(outfile + "-credmaster.txt"))
+		outfile = outfile + "-credmaster.txt"
```
Comment (knavesec) [inline credmaster.py:77]: remove the + "-credmaster" part, this will be added in log functions

File: credmaster.py [hunk 5]
```diff
@@ -148,18 +147,18 @@ def main(args,pargs):
 		key = pargs[i].replace("--","")
 		pluginargs[key] = pargs[i+1]
 
-	start_time = datetime.datetime.utcnow()
-	log_entry('Execution started at: {}'.format(start_time))
+	start_time = utc_now
```
Comment (knavesec) [inline credmaster.py:150]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 6]
```diff
@@ -230,8 +229,8 @@ def main(args,pargs):
 
 				weekdaywarrior = int(weekdaywarrior)
 				sleep_time = ww_calc_next_spray_delay(weekdaywarrior)
-				next_time = datetime.datetime.utcnow() + datetime.timedelta(hours=weekdaywarrior) + datetime.timedelta(minutes=sleep_time)
-				log_entry("Weekday Warrior, sleeping {delay} minutes until {time} on {day} in UTC {utc}".format(delay=sleep_time,time=next_time.strftime("%H:%M"),day=spray_days[next_time.weekday()], utc=weekdaywarrior))
+				next_time = utc_now + datetime.timedelta(hours=weekdaywarrior) + datetime.timedelta(minutes=sleep_time)
```
Comment (knavesec) [inline credmaster.py:232]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 7]
```diff
@@ -251,22 +250,22 @@ def main(args,pargs):
 
 			if delay is None or len(passwords) == 1 or password == passwords[len(passwords)-1]:
 				if userpass_file != None:
-					log_entry('Completed spray with user-pass file {} at {}'.format(userpass_file, datetime.datetime.utcnow()))
+					log_entry(f"Completed spray with user-pass file {userpass_file} at {utc_now}")
```
Comment (knavesec) [inline credmaster.py:253]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 8]
```diff
@@ -251,22 +250,22 @@ def main(args,pargs):
...
 				elif userenum:
-					log_entry('Completed userenum at {}'.format(datetime.datetime.utcnow()))
+					log_entry(f"Completed userenum at {utc_now}")
```
Comment (knavesec) [inline credmaster.py:255]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 9]
```diff
@@ -251,22 +250,22 @@ def main(args,pargs):
...
 				else:
-					log_entry('Completed spray with password {} at {}'.format(password, datetime.datetime.utcnow()))
+					log_entry(f"Completed spray with password {password} at {utc_now}")
```
Comment (knavesec) [inline credmaster.py:257]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 10]
```diff
@@ -251,22 +250,22 @@ def main(args,pargs):
...
 
 				notify.notify_update("Info: Spray complete.", notify_obj)
 				continue
 			elif count != passwordsperdelay:
-				log_entry('Completed spray with password {} at {}, moving on to next password...'.format(password, datetime.datetime.utcnow()))
+				log_entry(f"Completed spray with password {password} at {utc_now}, moving on to next password...")
```
Comment (knavesec) [inline credmaster.py:262]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 11]
```diff
@@ -251,22 +250,22 @@ def main(args,pargs):
...
 				continue
 			else:
-				log_entry('Completed spray with password {} at {}, sleeping for {} minutes before next password spray'.format(password, datetime.datetime.utcnow(), delay))
-				log_entry('Valid credentials discovered: {}'.format(len(results)))
+				log_entry(f"Completed spray with password {password} at {utc_now}, sleeping for {delay} minutes before next password spray")
```
Comment (knavesec) [inline credmaster.py:265]: Remove utc_now, else all timestamps will be exactly the same

File: credmaster.py [hunk 12]
```diff
@@ -285,7 +284,7 @@ def main(args,pargs):
 			log_entry("Second KeyboardInterrupt detected, unable to clean up APIs :( try the --clean option")
 
 	# Capture duration
-	end_time = datetime.datetime.utcnow()
+	end_time = utc_now
```
Comment (knavesec) [inline credmaster.py:287]: Remove utc_now, else all timestamps will be exactly the same

File: plugins/azuresso/azuresso.py [hunk 13]
```diff
@@ -77,46 +77,46 @@ def azuresso_authenticate(url, username, password, useragent, pluginargs):
...
             if (token):
-                data_response['output'] += " - GOT TOKEN {}".format(token[0])
+                data_response['output'] += f" - GOT TOKEN {token[0]}"
 
         elif "AADSTS50056" in xmlresponse:
-            data_response['output'] = "[!] VALID_USERNAME - {} (no password in AzureAD)".format(creds)
+            data_response['output'] = f"[!] VALID_USERNAME - {creds} (no password in AzureAD)"
             data_response['result'] = "failure"
-            data_response['valid_user'] = True
+            # utils.slacklog("Alert: Username valid but password is not in AzureAD")
```
Comment (knavesec) [inline plugins/azuresso/azuresso.py:106]: remove

File: plugins/azuresso/azuresso.py [hunk 14]
```diff
@@ -77,46 +77,46 @@ def azuresso_authenticate(url, username, password, useragent, pluginargs):
...
             token = re.findall(r"<DesktopSsoToken>.{1,}</DesktopSsoToken>", xmlresponse)
             if (token):
-                data_response['output'] += " - GOT TOKEN {}".format(token[0])
+                data_response['output'] += f" - GOT TOKEN {token[0]}"
 
         elif "AADSTS50056" in xmlresponse:
-            data_response['output'] = "[!] VALID_USERNAME - {} (no password in AzureAD)".format(creds)
+            data_response['output'] = f"[!] VALID_USERNAME - {creds} (no password in AzureAD)"
             data_response['result'] = "failure"
-            data_response['valid_user'] = True
```
Comment (knavesec) [inline plugins/azuresso/azuresso.py:107]: add back in

File: plugins/azuresso/azuresso.py (hunk 13 above)
Comment (knavesec) [reply plugins/azuresso/azuresso.py:106]: Do this for all plugins, notifications done outside plugins

File: plugins/azuresso/azuresso.py (hunk 14 above)
Comment (knavesec) [reply plugins/azuresso/azuresso.py:107]: Add back in valid user tags for all plugins, used for automated logging of valid users

File: plugins/msol/msol.py [hunk 15]
```diff
@@ -5,7 +5,7 @@ def msol_authenticate(url, username, password, useragent, pluginargs):
 
     data_response = {
         'result': None,    # Can be "success", "failure" or "potential"
-        'error': False,
+		'error' : False,
```
Comment (knavesec) [inline plugins/msol/msol.py:8]: Is it just me or are some of these randomly differently tabbed? Unsure how much that matters

File: plugins/template/template.py [hunk 16]
```diff
@@ -8,10 +8,9 @@ def template_authenticate(url, username, password, useragent, pluginargs): # CHA
     # error
     # output
     # success
-    # valid_user
```
Comment (knavesec) [inline plugins/template/template.py:11]: Add back in

**Review by knavesec — CHANGES_REQUESTED** [review]
Really appreciate you doing all this work to reformat the strings. Looks like there were some changes (new push) that may not have been replicated in the code that needs to be adjusted, but nothing too major

**Comment by ZephrFish** [comment]
Shall work on changes, will see if easier to implement into class implementation too

File: credmaster.py [hunk 17]
```diff
@@ -589,33 +582,18 @@ def log_entry(entry):
 
 	lock.acquire()
 
-	ts = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S.%f')[:-3]
-	print('[{}] {}'.format(ts, entry))
+	ts = utc_now.strftime('%Y-%m-%d %H:%M:%S.%f')[:-3]
```
[the remark this answers is not in the document]
Comment (ZephrFish) [reply credmaster.py:585]: the utc_now var is simplified, takes the old value and is assigned it shouldn't change the functionality from my testing

**Comment by ZephrFish** [comment]
Fixed changes as requested, will work on class re-work once these changes are committed :)

**Pull request closed without merging**